plans/benchmarks/benchmarks.go

```diff
@@ -164,3 +172,101 @@ func BarrierBench(runenv *runtime.RunEnv) error {
 
 	return nil
 }
+
+func SubtreePublishBench(runenv *runtime.RunEnv) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+	watcher, writer := sync.MustWatcherWriter(ctx, runenv)
+	defer watcher.Close()
+	defer writer.Close()
+
+	if err := sync.WaitNetworkInitialized(ctx, runenv, watcher); err != nil {
+		return err
+	}
+
+	var msg []byte
+	st := sync.Subtree{
+		GroupKey:    "messages",
+		PayloadType: reflect.TypeOf(&msg),
+		KeyFunc: func(val interface{}) string {
+			return string(*val.(*[]byte))
+		}}
+
+	seq, err := writer.Write(ctx, &st, runenv.TestRun)
+	if err != nil {
+		return err
+	}
+
+	// Create several tests up to 1MB
+	// Anything over 1500 is likely to have ethernet fragmentation
```

Comment:
Yeah, and 1mb is certainly an abusive usage of the sync service. I would test up to 64kb.